### PR TITLE
[OSPP] Proposal for Edge Intelligence Benchmark for Edge-Cloud Collaborative Lifelong Detection

### PR DESCRIPTION
**Pull request by iszhyang**
Proposal for Edge Intelligence Benchmark for Edge-Cloud Collaborative Lifelong Detection

As a very promising community in edge AI Benchmarking, ianvs still has the following problems to be solved

The automatic dataset download and management features are missing, and users still need tedious operations to use the dataset.
Repeated deployment of the edge-cloud collaboration environment is too heavy. The distributed collaborative system simulation feature of ianvs needs to be developed urgently.
This proposal provides features in datasets, edge-cloud collaborative AI system simulation, etc. for the current version of Ianvs.

This PR is a modification of the last [PR](https://github.com/kubeedge/ianvs/pull/24)

### CODE REVIEW

File: docs/proposals/simulation/simulation.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+#### Framework
+![img](images/simulation_framework.jpg)
+
+In the framwork of ianvs, the `simulation controller` is the core module of system simulation. The `simulation controller` has been supplemented, which build and deploy local edge-cloud simulation environment with K8s.
+
+![img](images/simulation_controller.jpg)
+
+The models in `simulation controller` are as follows:
+- The `Simulation System Administrator` is used to
+  1. parse the system config(simulation)
```
Comment (Poorunga) [inline docs/proposals/simulation/simulation.md:112]: Before creating a simulation environment, you need to check whether the environment has the startup conditions, such as whether tools such as docker and kind are installed, and whether the CPU and memory are sufficient.

Comment (Poorunga) [reply docs/proposals/simulation/simulation.md:112]: Your text description needs to be consistent with the description in the flowchart.

File: docs/proposals/simulation/simulation.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+1. start the benchmarkingjob
+2. check whether to start emulation based on `system config`
+3. `Simulation System Administrator` check environment and parse system config
+  - check environment: eg. check if the host has installed docker, kind, and whether the memory > 4GB
+  - parse simulation config
+4. `Simulation System Administrator` build the environment and create needed moudles.
+  - build the environment: deploying a cluster including K8s and GM (global manager) locally through [all-in-one scripts of sedna](https://github.com/kubeedge/sedna/blob/527c574a60d0ae87b0436f9a8b38cf84fb6dab21/docs/setup/all-in-one.md)
+  - create and deploy the `simulation job controller`
+5. `Simulation Job Administrator` generate and deploy simulation jobs.
+  - generate the YAML of simulation job base on `testenv.yaml` and `algorithm.yaml`
```
Comment (Poorunga) [inline docs/proposals/simulation/simulation.md:136]: Add a step here: we need to help algorithm developers build algorithms into docker images.

File: docs/proposals/simulation/simulation.md (hunk 1 above)
Comment (iszhyang) [reply docs/proposals/simulation/simulation.md:111]: Thanks. I have modified it according to your suggestion.

File: docs/proposals/simulation/simulation.md (hunk 2 above)
Comment (iszhyang) [reply docs/proposals/simulation/simulation.md:138]: Thanks. We do need to help users package the algorithms under test into docker image. I have made modifications to the process and flowchart.

**Review by MooreZheng — COMMENTED** [review]
Overall it looks good to me now. Major revision has been made compared with the last PR https://github.com/kubeedge/ianvs/pull/24

File: docs/proposals/simulation/simulation.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      # the url address of python file
+      url: "./examples/pcb-aoi/singletask_learning_bench/testenv/f1_score.py"
+```
+
+The expected implementation method is to modify `dataset.py` in the `testenvmanager` module as follows:
+
+1. Add `self.dataset_url`, `self.dataset_path`
+2. Add `download_save()` function
+
+![](https://github.com/iszhyang/images/blob/main/ianvs/dataset-download.png?raw=true)
```
Comment (jaypume) [inline docs/proposals/simulation/simulation.md:83]: It is suggest to put the image file to this repository, not a link to your private repository.

File: docs/proposals/simulation/simulation.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+Baseline algorithm: [RFNet](https://github.com/AHupuJR/RFNet)
+
+We will provide the cityscapes dataset and corresponding methods for ianvs. The specific operation is as follows:
+
+- dataset processing
+  - generate the index file for train and test
+- dataset download link
+  - provide the download link of cityscapes dataset in HUAWEI Cloud OBS.
+- Baseline algorithm
+  ![](https://github.com/iszhyang/images/blob/main/ianvs/rfnet.png?raw=true)
```
Comment (jaypume) [inline docs/proposals/simulation/simulation.md:97]: It is suggest to put the image file to this repository, not a link to your private repository.

Comment (iszhyang) [reply docs/proposals/simulation/simulation.md]: Thanks. I have changed the image links in proposal file to local repository.

**Comment by MooreZheng** [comment]
See simulator related issue, PR and demo:
- Issue: https://github.com/kubeedge/ianvs/issues/348
- Proposal Pull Request: https://github.com/kubeedge/ianvs/pull/35
- Code Pull Request: https://github.com/kubeedge/ianvs/pull/39
- Demo Video（in Chinese）on KEAW'22: https://www.bilibili.com/video/BV1nD4y1Y7cJ

**Pull request merged**